team-telnyx/telnyx-webrtc-flutter
# fix(WEBRTC-2118): Session ID and Gateway Request fixes

[WEBRTC-2118 - Session ID and Gateway Request fixes .](https://telnyx.atlassian.net/browse/WEBRTC-2118)

---

This PR aims to tackle a few reported issues. Previously we were waiting for a session ID to be provided by the backend after a successful login. This however meant that in some instances people would try to create calls when one had not been provided. This was made harder because of the tricky registration process (some people would assume they were registered when they weren't)

With this PR we now provide a session ID for the backend to use on login. (This is a change we have done on Android recently as well after a discussion with @StefanYohansson)
https://github.com/team-telnyx/telnyx-webrtc-android/pull/205

This PR also fixes the gateway registration in the instance of receiving FAIL_WAIT. The gateway counter was being reset causing an infinite loop of registrations when one failed. 

## :older_man: :baby: Behaviors
### Before changes
- SessionID was incorrectly named 
- No sessid being provided or set during log in (we were waiting for a backend response to set it first)
- Gateway request would infinitely retry when receiving FAIL_WAIT

### After changes
- Renamed SessionID (sessionId -> sessid)
- We now provide sessid as a login param for both token and credential logins
- _resetGatewayCounter is not being reset on the connect() method, but on success or final failure of registration

## ✋ Manual testing
1. Build and run app on Android, Web or iOS
2. Login with Credentials A
3. Open https://webrtc.telnyx.com/
4. Login there with Credentials B
5. Call between both clients and ensure all is working fine

For gateway request:
1. Purposefully use incorrect credentials
2. Retry should happen 3 times and then fail finally
